lib/_stream_transform.js

```diff
@@ -132,6 +132,8 @@ function Transform(options) {
 }
 
 function prefinish() {
+  if (this._readableState.destroyed)
+    return;
```

Comment:
can you add one more test for the `_flush()` logic and `destroy()`? It seems you are fixing a bug here. Or without this change some tests would not pass anymore?